packages/react-select/src/Select.js

```diff
@@ -1467,7 +1471,8 @@ export default class Select extends Component<Props, State> {
         form={form}
         theme={theme}
         type="text"
-        value={inputValue}
+        value={value ? value.value : inputValue}
```

Comment:
This would be a breaking change as existing behavior is to clear the input onBlur.